src/coreclr/jit/emitxarch.cpp

```diff
@@ -504,19 +504,18 @@ bool emitter::IsRexW1EvexInstruction(instruction ins)
     return false;
 }
 
-#ifdef TARGET_64BIT
 //------------------------------------------------------------------------
-// AreUpper32BitsZero: check if some previously emitted
-//     instruction set the upper 32 bits of reg to zero.
+// AreUpperBitsZero: check if some previously emitted
+//     instruction set the upper bits of reg to zero.
 //
 // Arguments:
 //    reg - register of interest
 //
 // Return Value:
-//    true if previous instruction zeroed reg's upper 32 bits.
+//    true if previous instruction zeroed reg's upper bits.
 //    false if it did not, or if we can't safely determine.
 //
-bool emitter::AreUpper32BitsZero(regNumber reg)
+bool emitter::AreUpperBitsZero(regNumber reg, emitAttr size)
```

Comment:
Add a description of the `size` argument above. E.g., if `size` is `EA_2BYTE`, does this function tell you that all register bits _above_ the low 2 bytes are zero? What if this is a 32-bit platform and you pass `EA_4BYTE` or higher; is that legal?

Reply:
Makes sense, will add a description.

Passing `EA_4BYTE` on 32-bit should be legal, but should always return false.